crypto/asn1/x_bignum.c

```diff
@@ -130,9 +130,24 @@ static int bn_c2i(ASN1_VALUE **pval, const unsigned char *cont, int len,
 static int bn_secure_c2i(ASN1_VALUE **pval, const unsigned char *cont, int len,
                          int utype, char *free_cont, const ASN1_ITEM *it)
 {
-    if (!*pval)
-        bn_secure_new(pval, it);
-    return bn_c2i(pval, cont, len, utype, free_cont, it);
+    int ret;
+    BIGNUM *bn;
+
+    if (!*pval) {
```

Comment:
It would be more stylish as `if (!*pval && !bn_secure_new(pval, it))`

Reply:
It would be more stylishly-correct to explicit test against NULL :)